### PR TITLE
satisfies?

### PR DESCRIPTION
**Pull request by lilactown**
-------

Please answer the following questions and leave the below in as part of your PR.

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code): [#issue-number](https://github.com/clavascript/clavascript/issues/...) (mention the issue number in the previous link and remove this text)
- [x] This PR provides tests for code additions and changes
- [x] This PR does not contain bonus changes, i.e. changes not discussed in the linked issue. For those, please submit a new issue + PR.

After submitting this PR, please do not force-push your branch as this makes
collaboration easier. Push as many new commits to your branch, they will be
squashed upon merge.

---

Implements `satisfies?`. Related #126

### CODE REVIEW

File: core.js [hunk 1]
```diff
@@ -7,6 +7,10 @@ export function _PLUS_(x, ...xs) {
   return sum;
 }
 
+export function satisfies_QMARK_(protocol, x) {
+  return protocol in x && x[protocol];
```
Comment (borkdude) [inline core.js:11]: Can you explain to me what the difference between the left hand and right hand side is?

Comment (lilactown) [reply core.js:11]: `protocol in x` is basically like `contains?` - it checks whether an object has a property in its.

`x[protocol]` returns the value at that property, which in the case of a properly implemented protocol should be `true`.

IDK, maybe I'm being a bit redundant.

Comment (borkdude) [reply core.js:11]: Wouldn't it be sufficient to return `x[protocol]`?

Comment (lilactown) [reply core.js:11]: yeah, that's what I ended up changing this to in https://github.com/clavascript/clavascript/pull/129

**Pull request merged**